chris_backend/plugininstances/services/manager.py

```diff
@@ -672,9 +672,39 @@ def _handle_finished_with_error_status(self):
         Internal method to handle the 'finishedWithError' status returned by the
         remote compute.
         """
-        self.c_plugin_inst.status = 'finishedWithError'
-        self.delete_plugin_instance_job_from_remote()
-        self.save_plugin_instance_final_status()
+        plg_inst_lock = PluginInstanceLock(plugin_inst=self.c_plugin_inst)
+        try:
+            plg_inst_lock.save()
+        except IntegrityError:
+            pass  # another async task has already entered here
```

Comment:
@jbernal0019 under what conditions would there be multiple async tasks executing the same code? I would think there is a better way to synchronize here.